### PR TITLE
[feat] Add connect wallet functionality

### LINKED ISSUE #110: [feat] Add connect wallet functionality
### This issue will be part of ODHack8.0, please apply via Onlydust app
- Add functionality to connect your starknet wallet, the page display the pop up and the user accept the terms and the wallet address will be return.
-  The connect wallet btn inside the app, should no longer be available, in its place, a partial string of the user's address should be displayed, ex: 0x123...321.
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue.

### PR DESCRIPTION
**Pull request by jimenezz22**
# Pull Request

- [x] Closes #110 
- [x] Added tests (if necessary)
- [x] Run tests
- [x] Run formatting
- [ ] Commented the code

## Changes description

Describe what your changes will do and any TODOs there might be in the future related to the issue you are closing.
- Use useState to manage wallet connection status (walletConnection), storing the wallet address and wallet object.
- The handleConnect function triggers wallet connection via connect() and retrieves the wallet's address.
- Upon successful connection, stores the address in the component state and persists it in localStorage for session persistence.
- The handleDisconnect function disconnects the wallet via disconnect(), clears the connection state, and removes relevant data from localStorage.
- The "Connect wallet" button dynamically changes to display the connected wallet’s shortened address (e.g., 0x1234...abcd) when a wallet is connected.
- A disconnect button allows the user to disconnect their wallet with a simple UI interaction.

## Current output

giphy

Before connecting:

Screenshot 2024-10-03 at 1 40 37 AM

During connecting:

Screenshot 2024-10-03 at 1 40 17 AM
Screenshot 2024-10-03 at 1 41 31 AM
Screenshot 2024-10-03 at 1 41 15 AM

After connecting:
Screenshot 2024-10-03 at 1 40 51 AM

## Time spent breakdown

Give a breakdown on the time you used to solve the issue you are closing.

- Approximately 4/5 hours divided into two days. I read the Starknetkit documentation required to connect the wallets, saw their examples and adapted it to what is required for this issue

## Comments

Any other details you might want to share.
- None

### CODE REVIEW

File: frontend/gostarkme-web/next.config.mjs
Comment (adrianvrj) [inline frontend/gostarkme-web/next.config.mjs:1]: please do not upload the changes for this file, we have to add it to the .gitignore actually lol

File: frontend/gostarkme-web/package.json
Comment (adrianvrj) [inline frontend/gostarkme-web/package.json:1]: please do not upload the changes for this file

File: frontend/gostarkme-web/yarn.lock
Comment (adrianvrj) [inline frontend/gostarkme-web/yarn.lock:1]: please do not upload the changes for this file

File: frontend/gostarkme-web/.env
Comment (adrianvrj) [inline frontend/gostarkme-web/.env:1]: please do not upload the changes for this file

**Review by adrianvrj — CHANGES_REQUESTED** [review]
Looks good, just a couple of changes

File: frontend/gostarkme-web/package.json
Comment (jimenezz22) [reply frontend/gostarkme-web/package.json:1]: @adrianvrj is important to use "starknet": "^6.7.0", and "starknetkit": "^1.1.5" as a versions to ensure the wallet connector funcionality

Comment (adrianvrj) [reply frontend/gostarkme-web/package.json:1]: I see, leave this change then, ty

**Review by adrianvrj — APPROVED** [review]

**Comment by adrianvrj** [comment]
@jimenezz22 seems like something went wrong when building the frontend

**Comment by jimenezz22** [comment]
> @jimenezz22 seems like something went wrong when building the frontend

checking

**Comment by jimenezz22** [comment]
@adrianvrj Everything looks good right now

Screenshot 2024-10-03 at 7 00 04 PM

**Comment by adrianvrj** [comment]
@jimenezz22 everything looks good now, tysm, great work homie!

**Pull request merged**